### PR TITLE
Initial support for compile with OTP-27

### PR DESCRIPTION
**Pull request by mths1**
## Proposed Changes

Initial support for compile with OTP-27. 
Notes:
* Nodetool works only in interpreted mode. OTP27 by default, compiles escripts. Unfortuntely, vernemq stop somehow crashes in compile mode. In interpreted mode, which was the default in OTP26 and earlier, it works as expected. We need to fix this for OTP28.
* code:lib_dir/2 has been deprecated. Needs to be solved in OTP29.

Things to come:
* JSON module support (follow up pull request). OTP27 does not need jsx anymore, and vmq_json can support both for a while and drop jsx with OTP29.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes issue #XXXX)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, styles...)
- [x] DevOps (Build scripts, pipelines...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [ ] I have read the [CODE_OF_CONDUCT.md](https://github.com/vernemq/vernemq/blob/main/CODE_OF_CONDUCT.md) document
- [ ] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if needed)
- [ ] Any dependent changes have been merged and published in related repositories
- [ ] I have updated changelog (At the bottom of the release version)
- [ ] I have squashed all my commits into one before merging

## Further Comments

If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc.

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -161,10 +161,13 @@ jobs:
           # Monterrey
           - macos-12
```
Comment (ioolkos) [inline .github/workflows/pr.yml:162]: Apparently macos-13 and macos-14 are available as Github runners. Should we switch to a newer macos here anyway?

File: rebar.config [hunk 2]
```diff
@@ -68,6 +68,8 @@
         vmq_server,
         vernemq_dev,
         credentials_obfuscation,
+        wx,
```
Comment (ioolkos) [inline rebar.config:71]: For a pure server build, we don't need `wx` but we could use multiple build profiles and have a profile for developing.

Comment (mths1) [reply rebar.config:71]: Oh, yeah. That is a left-over from debugging. Adding a profile which supports the debugger would be nice. Once I got it running, I have fallen in love with the gui debugger ;-)

Comment (ioolkos) [reply rebar.config:71]: it is cool!
to add a profile, we can add sth like the following to the Makefile:
```
dev_build: PROFILE = as develop
dev_build: rel
```

Then add `profiles` to the rebar.config (keeping the main relx build out of it):

```
{profiles, [
   {develop, [
        {relx, [
            {release, {vernemq, semver}, [
                sasl,
                inets,
                xmerl....]}]}]}
                
           ]}.
```
then build with     
`make dev_build`

**Pull request merged**